x-pack/plugins/observability/public/pages/alerts/alerts.tsx

```diff
@@ -58,7 +58,7 @@ function InternalAlertsPage() {
   } = useKibana().services;
   const { ObservabilityPageTemplate, observabilityRuleTypeRegistry } = usePluginContext();
   const alertSearchBarStateProps = useAlertSearchBarStateContainer(ALERTS_URL_STORAGE_KEY, {
-    replace: false,
+    replace: true,
```

Comment:
@jennypavlova Thanks for checking this issue. I initially had replace as `true`, but when we use different time ranges on the AlertSummaryWidget chart using the brush, we no longer have the history for time changes to go back. 
I should have a ticket for it, let me find and share it here.

I think we need to see why there is a double history in the URL and fix that instead of replacing the URL. Still, I am open to this option if designers think this approach is better, but it seems more like a technical issue rather than a UX one.

Reply:
@maryam-saeidi That's a good point that probably the brush history needs to be there on the back click. I am wondering if we still want to have 2 history elements when we come from a different page tho 🤔 I am checking if maybe in this case we can have the replace set to true if the previous page is not `/alerts` and set to false if the previous history element is the alerts page as a possible way to fix it.

Reply:
@maryam-saeidi Ignore the previous suggestion - it is also not a good option as we navigate using `RedirectAppLinks`. 
What I would suggest: Ater looking into the code inside the `syncUrlStateWithInitialContainerState` I saw several cases there. We have `false` as a valid case only if the parameters are not provided (so in the case when we use brush we change only the time series or default but we don't provide an extra query for example) we can set true only for the `isRight` case which is the case where we provide the parameters like our redirection case - that way the brush action will work as expected and the navigation with the provided parameters will also use replace to fix the existing bug. Wdyt?
Here is an example result:

https://github.com/elastic/kibana/assets/14139027/455bfa10-51fb-4afd-940b-1e1e1aacfaeb

Reply:
Looks good to me! 👍🏻